### PR TITLE
IF-290: Extend asset creation with subsystems

### PR DESCRIPTION
**Pull request by jsylupp**

### CODE REVIEW

**Comment by mattmikulina** [comment]
Task linked: [IF-290 Create Asset with subsystems (of owning assets)](https://app.clickup.com/t/2177764/IF-290)

File: fusionfrontend/src/app/components/ui/trash-button/trash-button.component.scss [hunk 1]
```diff
@@ -31,7 +31,6 @@
   cursor: pointer;
 
   margin: 8px 4px;
-  padding-top: 6px;
```
Comment (seemueller) [inline fusionfrontend/src/app/components/ui/trash-button/trash-button.component.scss:34]: colors should be variables

File: fusionfrontend/src/app/dashboards/components/content/maintenance-list/maintenance-list.component.ts [hunk 2]
```diff
@@ -154,19 +154,21 @@ export class MaintenanceListComponent implements OnInit, OnChanges {
...
+      this.displayedFactoryAssets = this.displayedFactoryAssets
+        .filter(asset => factorySiteNames.includes(asset.factorySiteName));
     }
     if (assetTypeNames.length > 0) {
-      this.displayedAssets = this.displayedAssets.filter(asset => assetTypeNames.includes(asset.category));
+      this.displayedFactoryAssets = this.displayedFactoryAssets.filter(asset => assetTypeNames.includes(asset.category));
     }
     if (companyNames.length > 0) {
-      this.displayedAssets = this.displayedAssets.filter(asset => companyNames.includes(asset.manufacturer));
+      this.displayedFactoryAssets = this.displayedFactoryAssets.filter(asset => companyNames.includes(asset.manufacturer));
```
Comment (seemueller) [inline fusionfrontend/src/app/dashboards/components/content/maintenance-list/maintenance-list.component.ts:171]: code copy extracted in method?

File: fusionfrontend/src/app/fleet/components/content/asset-wizard/asset-wizard-shared/asset-wizard-shared-subsystems/asset-wizard-shared-subsystems.component.scss [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+.asset-wizard-subsystems {
+  .asset-icon {
+    display: flex;
+    align-items: center;
+
+    img {
+      width: 2rem;
+      height: 2rem;
+      /*margin-top: 9px;*/
```
Comment (seemueller) [inline fusionfrontend/src/app/fleet/components/content/asset-wizard/asset-wizard-shared/asset-wizard-shared-subsystems/asset-wizard-shared-subsystems.component.scss:27]: no code comment

File: fusionfrontend/src/app/fleet/components/content/asset-wizard/asset-wizard.component.ts [hunk 4]
```diff
@@ -135,7 +138,7 @@ export class AssetWizardComponent implements OnInit {
 
   onSaveAsset(): void {
     if (this.asset && this.assetForm.valid && this.asset.fieldInstances
-        && this.metricsValid && this.attributesValid && this.customerDataValid) {
+        && this.metricsValid && this.attributesValid && this.subsystemsValid && this.customerDataValid) {
```
Comment (seemueller) [inline fusionfrontend/src/app/fleet/components/content/asset-wizard/asset-wizard.component.ts:141]: why is here getRawValue() not used?

File: fusionfrontend/src/assets/sass/components/_primedropdown.scss [hunk 5]
```diff
@@ -24,6 +24,8 @@
   align-items: center;
   box-shadow: none !important;
 
+/*  .p-inputtext, .p-dropdown .p-inputtextarea*/
```
Comment (seemueller) [inline fusionfrontend/src/assets/sass/components/_primedropdown.scss:27]: remove comment

**Review by seemueller — CHANGES_REQUESTED** [review]

**Pull request merged**